cli/azd/pkg/containerapps/container_app.go

```diff
@@ -124,6 +131,21 @@ func (cas *containerAppService) DeployYaml(
 		return fmt.Errorf("decoding yaml: %w", err)
 	}
 
+	if shouldPersist := cas.alphaFeatureManager.IsEnabled(persistCustomDomainsFeature); shouldPersist {
+		aca, err := cas.getContainerApp(ctx, subscriptionId, resourceGroupName, appName)
+		if err == nil {
+			acaAsConfig := config.NewConfig(obj)
```

Comment:
Slightly nervous that the semantics of `config.Config` interacting poorly with the JSON here, and I would have just considered setting this on the object itself, but I guess this saves you casting, so if you feel like this is safe I guess it is good.  I'm just worried this is going to rewrite parts of the JSON in some way we don't expect...

Reply:
`config.Config` uses `map[string]any` the same way as JSON, and the `.Set()` is super convenient because it creates the JSON path automatically